src/main/java/org/example/DAO/imp/LineItemDao.java

```diff
@@ -0,0 +1,138 @@
+package org.example.DAO.imp;
+
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.TransactionRequiredException;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.CriteriaUpdate;
+import javax.persistence.criteria.Root;
+import lombok.extern.slf4j.Slf4j;
+import org.example.DAO.Dao;
+import org.example.model.LineItem;
+
+/** Class responsible for database operations at lineItem entity. */
+@Slf4j
+public class LineItemDao implements Dao<LineItem> {
+  EntityManagerFactory etf;
+  EntityManager em;
+  CriteriaBuilder cb;
+
+  public LineItemDao(EntityManagerFactory etf) {
+    this.etf = etf;
+    em = etf.createEntityManager();
+    cb = em.getCriteriaBuilder();
+  }
+
+  /**
+   * Uses an entity manager to insert a lineItem into the database.
+   *
+   * @param lineItem the lineItem object who will be added at database.
+   */
+  @Override
+  public void add(LineItem lineItem) {
+    try {
+      em.getTransaction().begin();
+      em.persist(lineItem);
+    } catch (IllegalArgumentException | TransactionRequiredException e) {
+      log.error("Unable to add item to order.");
+    } finally {
+      em.getTransaction().commit();
+    }
+  }
+
+  /**
+   * Uses an entity manager to insert a lineItem into the database.
+   *
+   * @param id the lineItem id who will be searched at database.
+   * @return a lineItem object.
+   */
+  @Override
+  public LineItem get(Long id) {
+    LineItem lineItem = null;
+    try {
+      em.getTransaction().begin();
+      lineItem = em.find(LineItem.class, id);
+      if (lineItem == null) {
```

Comment:
Look into other ways of nullchecking. 

https://www.baeldung.com/java-avoid-null-check

Reply:
Done.